test/integration/federation/server_test.go

```diff
@@ -69,11 +76,32 @@ func TestRun(t *testing.T) {
 	if err := waitForApiserverUp(); err != nil {
 		t.Fatalf("%v", err)
 	}
+}
+
+// Verifies that only default APIs are enabled when no runtime config is set.
+func TestDefaultRun(t *testing.T) {
+	run(t, "")
+	expectedGroupVersions := enabledGroupVersions
 	testSwaggerSpec(t)
 	testSupport(t)
-	testAPIGroupList(t)
-	testAPIGroup(t)
-	testAPIResourceList(t)
+	testAPIGroupList(t, expectedGroupVersions)
+	testAPIGroup(t, expectedGroupVersions)
+	testAPIResourceList(t, expectedGroupVersions)
+}
+
+// Verifies that all APIs are enabled when runtime config is set to all.
+func TestRunWithRuntimeConfigAll(t *testing.T) {
+	securePort = 6443 + 5
```

Comment:
Or just cherry-pick https://github.com/kubernetes/kubernetes/pull/42278/commits/e51262888227119d1c2b1fe8c64cec209e254150 into your PR here and this is ready to go. Github will probably figure out the repeated commit itself.